### PR TITLE
Drop non-string labels if connected to older APM Server

### PR DESCRIPTION
**Pull request by felixbarny**
- [x] Implement code
- [x] Add tests
- [ ] ~Update documentation~
- [x] Update [CHANGELOG.md](CHANGELOG.md)
- [ ] ~Update [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)~

fixes elastic/apm-agent-java#687

### CODE REVIEW

**Comment by felixbarny** [comment]
run the tests

File: apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java [hunk 1]
```diff
@@ -64,34 +66,44 @@
...
-                    errorCount.set(0);
+                    setServerUrls(shuffleUrls(newValue));
                 }
             }
         });
-        this.serverUrls = Collections.unmodifiableList(shuffledUrls);
+        setServerUrls(Collections.unmodifiableList(shuffledUrls));
+    }
+
+    private void setServerUrls(List<URL> serverUrls) {
```
Comment (eyalkoren) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerClient.java]:
```suggestion
    private void initialize(List<URL> serverUrls) {
```

File: apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerHealthChecker.java [hunk 2]
```diff
@@ -24,36 +24,49 @@
...
 
+    @Nullable
     @Override
-    public void run() {
-        apmServerClient.executeForAllUrls("/", new ApmServerClient.ConnectionHandler<Void>() {
+    public Version call() {
+        List<Version> versions = apmServerClient.executeForAllUrls("/", new ApmServerClient.ConnectionHandler<Version>() {
             @Override
-            public Void withConnection(HttpURLConnection connection) {
+            public Version withConnection(HttpURLConnection connection) {
```
Comment (eyalkoren) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerHealthChecker.java]: What do you say about always returning a version from this method (eg version 0.0.0 on failures), so always having a list of versions at the size of configured server URLs? Then, we don't have to limit ourselves to look only at the minimal version, we can use the `errorCount` state to always know **the version of the currently connected server**.
We only need to make sure we use shuffle the URL list once and use the same shuffle here and for the other users.

Comment (eyalkoren) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/report/ApmServerHealthChecker.java]: Forget about it- not THAT required: if we have a bunch of APM servers with different versions (shouldn't be a typical case unless during an APM server cluster upgrade), we would probably want our agents to behave the same with all of them, so the minimal version is actually probably what we are looking for

**Pull request merged**